contracts/RandomNumberDirectFundingConsumerV2.sol

```diff
@@ -0,0 +1,129 @@
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+contract RandomNumberDirectFundingConsumerV2 is
+    VRFV2WrapperConsumerBase,
+    ConfirmedOwner
+{
+    event RequestSent(uint256 requestId, uint32 numWords, uint256 paid);
+    event RequestFulfilled(
+        uint256 requestId,
+        uint256[] randomWords,
+        uint256 payment
+    );
+    error RequestNotFound(uint256 requestId);
+    error LinkTransferError(address sender, address receiver, uint256 amount);
+
+    struct RequestStatus {
+        uint256 paid; // amount paid in link
+        bool fulfilled; // whether the request has been successfully fulfilled
+        uint256[] randomWords;
+    }
+    mapping(uint256 => RequestStatus)
+        public s_requests; /* requestId --> requestStatus */
+
+    // past requests Id.
+    uint256[] public requestIds;
+    uint256 public lastRequestId;
+
+    // Address LINK
+    address public immutable linkAddress;
+
+    // configuration: https://docs.chain.link/vrf/v2/direct-funding/supported-networks#configurations
+    constructor(
+        address _linkAddress,
+        address _wrapperAddress
+    )
+        ConfirmedOwner(msg.sender)
+        VRFV2WrapperConsumerBase(_linkAddress, _wrapperAddress)
+    {
+        linkAddress = _linkAddress;
+    }
+
+    // Depends on the number of requested values that you want sent to the
+    // fulfillRandomWords() function. Test and adjust
+    // this limit based on the network that you select, the size of the request,
+    // and the processing of the callback request in the fulfillRandomWords()
+    // function.
+    // The default is 3, but you can set this higher.
+    // For this example, retrieve 2 random values in one request.
+    // Cannot exceed VRFV2Wrapper.getConfig().maxNumWords.
+    function requestRandomWords(
+        uint32 _callbackGasLimit,
+        uint16 _requestConfirmations,
+        uint32 _numWords
+    ) external onlyOwner returns (uint256 requestId) {
+        requestId = requestRandomness(
+            _callbackGasLimit,
+            _requestConfirmations,
+            _numWords
+        );
+        uint256 paid = VRF_V2_WRAPPER.calculateRequestPrice(_callbackGasLimit);
+        s_requests[requestId] = RequestStatus({
+            paid: paid,
+            randomWords: new uint256[](0),
+            fulfilled: false
+        });
+        requestIds.push(requestId);
+        lastRequestId = requestId;
+        emit RequestSent(requestId, _numWords, paid);
+        return requestId;
+    }
+
+    function fulfillRandomWords(
+        uint256 _requestId,
+        uint256[] memory _randomWords
+    ) internal override {
+        RequestStatus memory request = s_requests[_requestId];
+        if (request.paid == 0) revert RequestNotFound(_requestId);
+        request.fulfilled = true;
+        request.randomWords = _randomWords;
+        s_requests[_requestId] = request;
```

Comment:
instead of pulling `s_requests[_requestId]` into memory, then assigning back to storage, why not just pull it from storage?

```
RequestStatus storage request = s_requests[_requestId];
if (request.paid == 0) revert RequestNotFound(_requestId);
request.fulfilled = true;
request.randomWords = _randomWords;
emit RequestFulfilled(_requestId, _randomWords, request.paid);
```